src/ucp/stream/stream_recv.c

```diff
@@ -124,27 +168,35 @@ ucp_stream_rdata_unpack(void *rdata, size_t length, ucp_request_t *dst_req)
     return status;
 }
 
+static UCS_F_ALWAYS_INLINE void
+ucp_stream_rdesc_advance(ucp_recv_desc_t *rdesc, size_t offset)
+{
+    ucp_stream_am_data_t *am_data;
+
+    ucs_assert(offset < rdesc->length);
+
+    rdesc->length        -= offset;
+    rdesc->stream_offset += offset;
+
+    am_data         = ucp_stream_rdesc_am_data(rdesc);
+    am_data->offset = rdesc->stream_offset + ucs_offsetof(ucp_stream_am_data_t,
```

Comment:
why do we need to save it here? it should be only in ucp_stream_recv_data_nb(). just before returning a descriptor.